snuba/settings_base.py

```diff
@@ -7,10 +8,13 @@
 
 PORT = 1218
 
+DEFAULT_DATASET_NAME = 'events'
+DISABLED_DATASETS = {}
+DATASET_MODE = 'DEV'
```

Comment:
I tried first to rely on application.testing and application.debug instead of introducing a new setting but I am not sure the two existing ones would have been a good idea since application.debug does not really mean "use dev table". I would say we may have to occasionally do some troubleshooting in debug mode in prod or vice versa, run prod mode on a laptop. I think having a separate setting is safer.

Reply:
@tkaemming what about this instead:
- we get rid of the DEBUG setting.
- we introduce a DEV setting with the expectations that it puts Snuba in "dev" mode instead of just activating debug mode on Flask
- we use this new setting to drive both the table name and the debug mode.

I think, this way, it would be less risky since people would not expect a "DEBUG" parameter to just turn on Flask DEBUG mode, and still we control everything with one flag.
What do you think ?